### PR TITLE
Issue25 date time pickers

### LINKED ISSUE #25: Complete date and time picker frontend
Add date and time picker dialogs to the create new todo dialog and unify the functions for both the create new and edit existing todos.

### Acceptance Criteria

- [x] Clicking on the time button on either the create new todo dialog or the existing todo cards opens up the time picker dialog

- [x] Clicking on the calendar button on either the create new todo dialog or the existing todo cards opens up the date picker dialog

- [x] The pickers for both create new todo and existing todos are calling the functions

- [x] Changes made to either the date or time pickers are reflected in the layout of the proper todo (whether that be the create new todo dialog or a current todo)

### LINKED ISSUE #28: Add backend to update fields
### Acceptance Criteria
- [x] - Add update queries for todo title, date, time, notes, and isCompleted
- [x] - Add interface to access the update queries

### PR DESCRIPTION
**Pull request by dansjack**
# Relevant Issues
resolves #25 
resolves #28 

# Summary
* Adds update queries to the DAO
* Adds a method to the Repository to route the different update types (date, time, title, etc) to the appropriate DAO query. Adds associated method in ViewModel to call the Repository method.
* Adds an interface to implement the ViewModel's updateTodo method.
* Adds class for DatePicker and TimePicker methods
* Attaches DatePicker and TimePicker to the Create Todo dialog
* You can now add a new todo with a date and time.
* Data in todos now persists after creating a new todo

### CODE REVIEW

File: app/src/main/java/com/ad340/whatdo/MainActivity.java [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
 
 import java.util.Calendar;
-import java.util.List;
 import java.util.Locale;
 
-public class MainActivity extends AppCompatActivity {
+import static com.ad340.whatdo.PickerUtils.showDatePicker;
+import static com.ad340.whatdo.PickerUtils.onDateSetListener;
+import static com.ad340.whatdo.PickerUtils.onTimeSetListener;
+import static com.ad340.whatdo.PickerUtils.showTimePicker;
```
Comment (lbrown51) [inline app/src/main/java/com/ad340/whatdo/MainActivity.java:28]: I have no change requested on this, but it's so weird that you need to import them.

File: app/src/main/java/com/ad340/whatdo/PickerUtils.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            c.set(Calendar.HOUR_OF_DAY, hourOfDay);
+            c.set(Calendar.MINUTE, minute);
+
+            SimpleDateFormat sdf = new SimpleDateFormat("h:mm a", Locale.US);
+            timeString.append(sdf.format(c.getTime()));
+        };
+    }
+
+
+    public static void showTimePicker(Context context, Calendar c, ImageButton imageButton,
```
Comment (lbrown51) [inline app/src/main/java/com/ad340/whatdo/PickerUtils.java:81]: Any thoughts on changing the showPicker method names to setTimePickerShowOnClick or something like that? Cause it seems like it's just the tiniest bit confusing to have a method called show that actually attaches an onClickListener. Is that just me?

File: app/src/main/java/com/ad340/whatdo/TodoRepository.java [hunk 3]
```diff
@@ -19,6 +19,29 @@
 
     LiveData<List<Todo>> getAllTodos() { return allTodos; }
 
+    void updateTodo(Todo todo, String data, int type) {
```
Comment (lbrown51) [inline app/src/main/java/com/ad340/whatdo/TodoRepository.java:22]: Clever!

File: app/src/main/java/com/ad340/whatdo/PickerUtils.java (hunk 2 above)
Comment (dansjack) [reply app/src/main/java/com/ad340/whatdo/PickerUtils.java:87]: Happy to go with whatever is easiest to understand for all involved! I'll change it 👌

Comment (dansjack) [reply app/src/main/java/com/ad340/whatdo/PickerUtils.java]: Updated

File: app/src/main/java/com/ad340/whatdo/MainActivity.java (hunk 1 above)
Comment (dansjack) [reply app/src/main/java/com/ad340/whatdo/MainActivity.java]: 😭

**Pull request merged**